tests/stub/configuration_hints/test_connection_recv_timeout_seconds.py

```diff
@@ -68,7 +68,7 @@ def _assert_is_timeout_exception(self, e):
     def _assert_is_client_exception(self, e):
         if get_driver_name() in ["java"]:
             self.assertEqual(
-                "org.neo4j.driver.exceptions.ClientException",
+                "org.neo4j.driver.exceptions.TransactionTerminatedException",
                 e.errorType
             )
         elif get_driver_name() in ["ruby"]:
```

Comment:
Similarly with `_assert_is_client_exception` and `_assert_is_tx_terminated_exception` in `tests/stub/tx_run/test_tx_run.py` which are newly added.

Reply:
The comparison logic was copied from some other tests, unfortunately I don't remember which. 🤔  I will fix these.

> Not sure if that should go into a separate PR

The `TransactionTerminatedException` is a new type that is introduced in a driver fix that depends on this PR, so I thought it would make sense to have it here.